File: Level1/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    basket=[] #뽑은 인형을 담는 리스트
       
    for move in moves: #하나씩 moves의 원소를
        for n in board: #행의 갯수만큼 반복
            if n[move-1] ==0: #비어있으면 다음 행으로
                continue
            elif n[move-1] !=0: #인형발견
                basket.append(n[move-1]) #바구니에 담는다.
                if len(basket)>1 and basket[-1]==basket[-2]: #만약 바구니에 쌓였을 때 연속으로 같은게 2개 있으면
                    basket.pop(-1)
                    basket.pop(-1)
                    answer+=2
                n[move-1]=0
                break
              
    return answer

File: Level1/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    stack<int> st;

    //명령어 확인
    for (int i = 0; i < moves.size(); ++i) {
        int cmd = moves[i];
        
        //열로 이동
        for (int j = 0; j < board.size(); ++j) {
            if (board[j][cmd] != 0) {
                if (!st.empty() && st.top() == board[j][cmd])
                    answer += 2;
                else
                    st.push(board[j][cmd]);
                board[j][cmd] = 0;
                break;
            }
        }
    }

    return answer;
}
